applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp

```diff
@@ -0,0 +1,299 @@
+const void Define2DWakeProcess::MarkWakeElements()
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    bool potentially_wake = false;
+    bool is_wake_element = false;
+    BoundedVector<double, 3> nodal_distances_to_wake = ZeroVector(3);
+
+    #pragma omp parallel for private(potentially_wake, is_wake_element, nodal_distances_to_wake)
+    for (int i = 0; i < static_cast<int>(root_model_part.Elements().size()); i++) {
+        ModelPart::ElementIterator it_elem = root_model_part.ElementsBegin() + i;
+
+        // Check if the element is touching the trailing edge
+        CheckIfTrailingEdgeElement(it_elem);
+
+        // Elements downstream the trailing edge can be wake elements
+        potentially_wake = CheckIfPotentiallyWakeElement(it_elem);
+
+        if (potentially_wake) {
+            // Compute the nodal distances of the element to the wake
+            nodal_distances_to_wake = ComputeNodalDistancesToWake(it_elem);
+
+            // Selecting the cut (wake) elements
+            is_wake_element = CheckIfWakeElement(nodal_distances_to_wake);
+
+            // Mark wake element and save their nodal distances to the wake
+            if (is_wake_element) {
+                #pragma omp critical
+                {
+                    it_elem->SetValue(WAKE, true);
+                    it_elem->SetValue(ELEMENTAL_DISTANCES, nodal_distances_to_wake);
+                    for (unsigned int i = 0; i < it_elem->GetGeometry().size(); i++) {
+                        it_elem->GetGeometry()[i].FastGetSolutionStepValue(DISTANCE) = nodal_distances_to_wake(i);
+                    }
+                }
+            }
+        }
+    }
+    // Add the trailing edge elements to the trailing_edge_sub_model_part
+    AddTrailingEdgeElements();
+}
+
+// This function checks if the element is touching the trailing edge
+const void Define2DWakeProcess::CheckIfTrailingEdgeElement(ElementIteratorType& rElement)
+{
+    // Loop over element nodes
+    for (unsigned int i = 0; i < rElement->GetGeometry().size(); i++) {
+        // Elements touching the trailing edge are trailing edge elements
+        if (rElement->GetGeometry()[i].Id() == mTrailingEdgeNode->Id()) {
+            MarkTrailingEdgeElement(rElement);
+        }
+    }
+}
+
+// This function marks the elements touching the trailing edge as trailing edge
+const void Define2DWakeProcess::MarkTrailingEdgeElement(ElementIteratorType& rElement)
+{
+    #pragma omp critical
+    {
+    rElement->SetValue(TRAILING_EDGE, true);
+    mTrailingEdgeElementsOrderedIds.push_back(rElement->Id());
+    }
+}
+
+// This function selects the elements downstream the trailing edge as
+// potentially wake elements
+const bool Define2DWakeProcess::CheckIfPotentiallyWakeElement(ElementIteratorType& rElement)
+{
+    // Compute the distance from the trailing edge to the element's center
+    BoundedVector<double, 3> distance_to_element_center = ZeroVector(3);
+    distance_to_element_center =
```

Comment:
```suggestion
    const auto distance_to_element_center =
```
And you can remove the declaration in the line above.